app-modules/division/src/Observers/DivisionObserver.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Assist\Division\Observers;
+
+use Assist\Division\Models\Division;
+
+class DivisionObserver
+{
+    public function creating(Division $division): void
+    {
+        $division->createdBy()->associate($division->createdBy ?? auth()->user());
+        $division->updatedBy()->associate($division->updatedBy ?? auth()->user());
```

Comment:
What happens here if this is triggered via automated processes and there is no authed User? Do we need protections to ensure that this only runs if `auth()->user()` is not null?

Reply:
Disassociates as per our discussion.